src/modules/musig/main_impl.h

```diff
@@ -548,10 +605,16 @@ int secp256k1_musig_partial_sig_verify(const secp256k1_context* ctx, const secp2
     if (!secp256k1_xonly_pubkey_load(ctx, &rp, &signer->nonce)) {
         return 0;
     }
+
     /* If the MuSig-combined point has an odd Y coordinate, the signers will
      * sign for the negation of their individual xonly public key such that the
-     * combined signature is valid for the MuSig aggregated xonly key. */
-    if (session->pre_session.pk_parity) {
+     * combined signature is valid for the MuSig aggregated xonly key. If the
+     * MuSig-combined point was tweaked then `e` is negated if the combined key
+     * has an odd Y coordinate XOR the internal key has an odd Y coordinate.*/
+    if (session->pre_session.pk_parity
+            + (session->pre_session.is_tweaked
+                && session->pre_session.internal_key_parity)
+            % 2 == 1) {
```

Comment:
This made me laugh but it is correct.

It could maybe be more clearly written
```
if (session->pre_session.pk_parity != (session->pre_session.is_tweaked && session->pre_session.internal_key_parity))
```

Reply:
Oh, I meant to comment on the earlier instance of `% 2 == 1`, where it makes more sense that you did this. Then this one makes sense for consistency's sake.

Reply:
`!=` is certainly more idiomatic

Reply:
even with consistency in mind `% 2 == 1` looks crappy. Switched to the `!=` version.